src/dispatch/incident/severity/service.py

```diff
@@ -91,6 +94,10 @@ def get_by_name_or_default(
 ) -> IncidentSeverity:
     """Returns an incident severity based on a name or the default if not specified."""
     if incident_severity_in:
+        if not incident_severity_in.project or incident_severity_in.project.id != project_id:
```

Comment:
If I'm understanding correctly, I think these should be exceptions as the caller has clearly become confused e.g. the `incident_in.project_id` differs with `project_id`.

Reply:
I went with the less strict approach due to the doc string. 
My interpretation of the doc string was that in the event multiple projects have severities with the same name, an incident's severity should resolve to the one within its project.

I'm pretty amenable to making these constraints more strict though. How strict do we want to enforce these project/incident/case-level separations moving forward?